be/src/storage/tablet_manager.cpp

```diff
@@ -136,30 +136,12 @@ Status TabletManager::create_tablet(const TCreateTabletReq& request, std::vector
     LOG(INFO) << "Creating tablet " << tablet_id;
 
     std::unique_lock wlock(_get_tablets_shard_lock(tablet_id));
-    // Make create_tablet operation to be idempotent:
-    // 1. Return true if tablet with same tablet_id and schema_hash exist;
-    //           false if tablet with same tablet_id but different schema_hash exist.
-    // 2. When this is an alter task, if the tablet(both tablet_id and schema_hash are
-    // same) already exist, then just return true(an duplicate request). But if
-    // tablet_id exist but with different schema_hash, return an error(report task will
-    // eventually trigger its deletion).
-    if (request.tablet_schema.keys_type == TKeysType::PRIMARY_KEYS) {
-        TabletSharedPtr tablet = _get_tablet_unlocked(tablet_id, true, nullptr);
-        if (tablet != nullptr && tablet->tablet_state() != TABLET_SHUTDOWN) {
-            return Status::OK();
-        } else if (tablet != nullptr) {
-            return Status::InternalError("primary key tablet still resident in shutdown queue");
-        }
-    } else if (_check_tablet_id_exist_unlocked(tablet_id)) {
```

Comment:
If the _check_tablet_id_exist_unlocked is removed in this place, the function is useless.
You can remove the function.

Reply:
removed, PTAL